pandas/io/json/_normalize.py

```diff
@@ -148,13 +149,13 @@ def _normalise_json(
     if isinstance(data, dict):
         for key, value in data.items():
             new_key = f"{key_string}{separator}{key}"
+
+            if not key_string:
+                new_key = removeprefix(new_key, separator)
```

Comment:
this is only available in python3.8 and under - this is on purpose, because it "forces" you to write it like
```suggestion
                if sys.version_info < (3, 9):
                    from pandas.util._str_methods import removeprefix
                    new_key = removeprefix(new_key, separator)
                else:
                    new_key = new_key.removeprefix(separator)
```
and then when Python3.8 is dropped, pyupgrade will rewrite this automatically to only keep
```
new_key = new_key.removeprefix(separator)
```
(you can see what will happen with `pyupgrade pandas/io/json/_normalize.py --py39-plus`)

Reply:
Ah gotcha - sorry misunderstood before thought that compat was handled directly in pandas.util._str_methods